### PR TITLE
fix breaking pathlib change in 3.10 beta1

### PR DESCRIPTION
**Pull request by carlmontanari**
Hello!

Sorry for the PR outa the blue w/out an issue to discuss -- had some time and figured I'd take a crack at it. I added 3.10 beta1 to my ci for some projects using pyfakefs and my builds are failing. Did a bit of sleuthing and it seems there were a few changes to Pathlib for this release. The main issue/thread seems to be [here](https://github.com/python/cpython/pull/19342) -- and associated [commit](https://github.com/python/cpython/commit/2219187cab6bca009c42b63b2f4c30b5b10c916d). This [commit](https://github.com/python/cpython/commit/abf964942f97f6489360a75fd57b5e4f41c75f57) also appears to be related.

I *think* (hope?!) that this fix is backwards compatible. Tests in the docker container all seem happy, and I've tested this locally on my Mac w/ 3.10 beta1 and the tests that are/were failing in CI pass like normal. For reference, [here](https://github.com/scrapli/scrapli_cfg/runs/2547508032?check_suite_focus=true) is my failed tests (the `build_posix (ubuntu-latest, 3.10.0-beta.1)` job). 

The fix itself is basically to ignore that `init` argument (the pr/issue on Cpython seems to say this is mostly legacy and (obviously) going away anyway). Then allowing the wrapped stat method (well I guess everything... which is perhaps not ideal?) to accept kwargs as well. Since `_init` method is no longer called, I removed that and just assigned that `_accessor` to the `_fake_accessor` directly after the `_from_parts` calls. 

Hope that all makes sense -- have been happily using pyfakefs (thanks for the project btw!) a while but the guts of it are probably a bit over my head :)

Carl

### CODE REVIEW

File: pyfakefs/fake_pathlib.py
Comment (carlmontanari) [reply pyfakefs/fake_pathlib.py:725]: Fixed!

File: .github/workflows/pythonpackage.yml
Comment (mrbean-bremen) [inline .github/workflows/pythonpackage.yml:32]: Thanks! I think you need `3.10-dev` here - GH actions won't understand that.

Comment (mrbean-bremen) [reply .github/workflows/pythonpackage.yml:32]: > looks like we may need to update the python actions to v2 as well

Ah, didn't realize that, thanks! Though currently some tests are failing for all versions, something is not correct yet...

Comment (mrbean-bremen) [reply .github/workflows/pythonpackage.yml:32]: >  I'll try to take a peak at the failures in later today/tomorrow

Thanks - I can also have another look if you want, but probably not before tomorrow evening (noon for you).

**Comment by carlmontanari** [comment]
Just pushed a commit that restores the `_init` method.... for whatever reason it seems that its working. I would have assumed because it was being called in `__new__` we would have been hitting these lines where we set the accessor and closed anyway?   Can circle back on this tomorrow with fresh eyes if that didn't work for some reason :D

File: pyfakefs/fake_filesystem.py
Comment (carlmontanari) [inline pyfakefs/fake_filesystem.py:1185]: this looks like another update that was made to pathlib (https://github.com/python/cpython/blob/main/Lib/pathlib.py#L1052)

not sure this is a particularly good way to handle it but it *does* seem to work

File: pyfakefs/fake_pathlib.py
Comment (carlmontanari) [inline pyfakefs/fake_pathlib.py:145]: this and the above section I am 99% sure I am doing something wrong/bad here -- but the gist of this is it looks like we needed to add some methods to the FakeAccessor object -- I am *assuming*(?) its not a great idea to just attach `os` methods here because we would want to have the accessor using the patched methods? But this does seem to get more of the tests passing so figured id commit it to try to be as helpful as I can... (even if it isn't much :))

**Comment by carlmontanari** [comment]
Last push for the day because I am def out of my element here :) I left some comments in the files changed section about things im not super sure about and/or I think are probably wrong. This does get tox tests passing locally but some of it feels fairly wrong. Hopefully this can at least save ya some time on figuring stuff out!

**Comment by carlmontanari** [comment]
@mrbean-bremen Sorry I didn't get a chance to look today either, and sorry for my barrage of not working commits :). If you dont get to it first I will try to spend some more time looking at this this weekend but I think I am in over my head! Whatever is easiest for you w/ respect to the PR I'm totally good w/! I'm glad to have at least helped a little bit!

**Comment by mrbean-bremen** [comment]
@carlmontanari -- thanks! There is no need for you to spend more time on this, I can take it from here.  As I wrote, your contribution has already been helpful enough. I'll work on your branch then if you don't mind and will see if I can finish it soon -- maybe at the weekend.

File: pyfakefs/fake_filesystem.py
Comment (mrbean-bremen) [reply pyfakefs/fake_filesystem.py:1185]: Will leave it as is, I think it is good enough for now.

File: pyfakefs/fake_pathlib.py
Comment (mrbean-bremen) [reply pyfakefs/fake_pathlib.py:149]: I changed it accordingly. Regarding `getcwd` - if using `os` inside the function, as I do it now, it will be patched, so it will behave correctly.

**Comment by mrbean-bremen** [comment]
@carlmontanari - I got it to work now, I'll let you have a look and check if something is missing from your POV before I'm going to merge it. Didn't have to wait until the weekend, as today was a holiday here...

File: pyfakefs/fake_pathlib.py
Comment (mrbean-bremen) [reply pyfakefs/fake_pathlib.py]: This change was actually not needed - it happened because the fake accessor was not set (why I had to implement the two class methods to call `_init`).

File: pyfakefs/fake_pathlib.py
Comment (mrbean-bremen) [reply pyfakefs/fake_pathlib.py:149]: I removed `realpath` as it doesn't seem to be really needed, but maybe I'm wrong here.

**Comment by carlmontanari** [comment]
> @carlmontanari - I got it to work now, I'll let you have a look and check if something is missing from your POV before I'm going to merge it. Didn't have to wait until the weekend, as today was a holiday here...

You rock! A quick check and looks like all my tests are passing on 3.10 locally. Will try to get a branch updated to remove my skips to let it run on 3.10 on ubuntu latest too (locally am macOS).

**Review by mrbean-bremen — APPROVED** [review]

**Pull request merged**
